### ISSUE TITLE
strictExportDeclareModifiers gives incorrect error message

### ISSUE THREAD

**Issue by sandersn** [Microsoft/dtslint#202]
See DefinitelyTyped/DefinitelyTyped#32884 for the correct repro. It says that `declare` is redundant in

```ts
declare namespace X {
}
export { X };
```

but it should actually say that `export` is the correct keyword. The error doesn't happen with `export = X` because there is a special exemption.

**Comment by sandersn** [comment]
Probably a duplicate of #184
